### PR TITLE
Fixed Any Type Collisions

### LINKED ISSUE #4306: Any Scalar for object does not work
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Cannot use AnyType when property an type of object.

Cannot use @serializationType(name: "global::System.Object")  in schema.extensions.graphql file as its not in StrawberryShake.CodeGeneration.CSharp.JsonUtils switch statement.
namespace StrawberryShake.CodeGeneration.CSharp
{
    public static class JsonUtils
    {
        public static string GetParseMethod(RuntimeTypeInfo serializationType)
        {
            return serializationType.ToString() switch
            {
                TypeNames.String => nameof(JsonElement.GetString),
                TypeNames.Uri => nameof(JsonElement.GetString),
                TypeNames.Byte => nameof(JsonElement.GetByte),
                TypeNames.ByteArray => nameof(JsonElement.GetBytesFromBase64),
                TypeNames.Int16 => nameof(JsonElement.GetInt16),
                TypeNames.Int32 => nameof(JsonElement.GetInt32),
                TypeNames.Int64 => nameof(JsonElement.GetInt64),
                TypeNames.UInt16 => nameof(JsonElement.GetUInt16),
                TypeNames.UInt32 => nameof(JsonElement.GetUInt32),
                TypeNames.UInt64 => nameof(JsonElement.GetUInt64),
                TypeNames.Single => nameof(JsonElement.GetSingle),
                TypeNames.Double => nameof(JsonElement.GetDouble),
                TypeNames.Decimal => nameof(JsonElement.GetDecimal),
                TypeNames.DateTimeOffset => nameof(JsonElement.GetString),
                TypeNames.DateTime => nameof(JsonElement.GetString),
                TypeNames.TimeSpan => nameof(JsonElement.GetString),
                TypeNames.Boolean => nameof(JsonElement.GetBoolean),
                TypeNames.Guid => nameof(JsonElement.GetGuid),
                _ => throw new NotSupportedException("Serialization format not supported.")
            };
        }

        public static string GetWriteMethod(RuntimeTypeInfo serializationType)
        {
            return serializationType.ToString() switch
            {
                TypeNames.String => nameof(Utf8JsonWriter.WriteString),
                TypeNames.Uri => nameof(Utf8JsonWriter.WriteString),
                TypeNames.Byte => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.ByteArray => nameof(Utf8JsonWriter.WriteBase64String),
                TypeNames.Int16 => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.Int32 => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.Int64 => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.UInt16 => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.UInt32 => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.UInt64 => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.Single => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.Double => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.Decimal => nameof(Utf8JsonWriter.WriteNumber),
                TypeNames.DateTimeOffset => nameof(Utf8JsonWriter.WriteString),
                TypeNames.DateTime => nameof(Utf8JsonWriter.WriteString),
                TypeNames.TimeSpan => nameof(Utf8JsonWriter.WriteString),
                TypeNames.Boolean => nameof(Utf8JsonWriter.WriteBoolean),
                TypeNames.Guid => nameof(Utf8JsonWriter.WriteString),
                _ => throw new NotSupportedException("Serialization format not supported.")
            };
        }
    }
}

### Steps to reproduce

Step 1: Open documentationhttps://chillicream.com/docs/strawberryshake/scalars#any-or-json
Step 2. Create a model with and propery of type object.
public class Composite {
  public object Sub {get;set;}
}

type Composite {
     sub: Any
}

Step 3:Extend scalar Any
  @serializationType(name: "global::System.Object")
  @runtimeType(name: "global::System.Text.Json.JsonElement")
in Serialization format not supported

Step 4: Build project and get error Serialization format not supported

### Relevant log output

```shell
Serialization format not supported
```

### Additional Context?

_No response_

### Product

Strawberry Shake

### Version

12.0.1

### LINKED ISSUE #4405: Can't use schema with Any scalar type
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am using StrawberryShake to access a HotChocolate GraphQL server that uses the Any scalar type. I have followed the guide on https://chillicream.com/docs/strawberryshake/scalars/#any-or-json to make this work.

When running `dotnet build` I get:

```
CSC : error SS0006: Unable to resolve type reference `None: Any`. [.....csproj]
```

When I replace Any with JSONObject in my schema file, the error goes away. It seems to be related to the name Any.

### Steps to reproduce

1. Set up a StrawberryShake client as described on https://chillicream.com/docs/strawberryshake/get-started/console, using a GraphQL endpoint that exposes the `Any` type.
2. Build the project using `dotnet build`

### Relevant log output

```shell
CSC : error SS0006: Unable to resolve type reference `None: Any`. [.....csproj]
```

### Additional Context?

_No response_

### Product

Strawberry Shake

### Version

12.1.0

### PR DESCRIPTION
**Pull request by jorrit**
This demonstrates the error described in #4405. I've narrowed my HotChocolate generated schema down to the essential difference that generates the error: when another scalar is defined, `Any` stops working.

The test failure goes away if I remove `scalar SomeScalar` or if I change `Any` to `JSONObject` (or anything else besides `Any`). There seems to be something weird going on with `Any`.

Closes #4405 
Closes #4306

### CODE REVIEW

**Comment by michaelstaib** [comment]
I will have a look.

**Comment by jorrit** [comment]
Thanks!

File: src/StrawberryShake/CodeGeneration/src/CodeGeneration/Utilities/SchemaHelper.cs [hunk 1]
```diff
@@ -72,6 +72,10 @@ public static class SchemaHelper
                                 scalar.Name.Value,
                                 scalar.Description?.Value));
                         }
+                        else if (scalar.Name.Value == ScalarNames.Any)
+                        {
+                            builder.AddType(new AnyType());
```
Comment (PascalSenn) [inline src/StrawberryShake/CodeGeneration/src/CodeGeneration/Utilities/SchemaHelper.cs:77]: @michaelstaib  so the problem of this sits deep in the type system i think. 
We add all unkonw scalars as `AnyType`. The Typesystem somehow then thinks it knows the `AnyType` already when there is another scalar defined. I tried to find a way to fix this in the core, but i feel like i am running in circles and looking in the wrong places. 
Anyway, adding the AnyScalar explicitly fixes this particular issue. But this is something we have to keep in mind for stitching as well

**Comment by PascalSenn** [comment]
@jorrit  This is now fixed in the newest version 12 and preview 13.

**Pull request closed without merging**